### PR TITLE
Add missing packages to toolchain

### PR DESCRIPTION
**Pull request by anphel31**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
A few toolchain packages have been rebuilding outside of the toolchain build process (in the package build phase). This should never happen, since the environments can differ. An analysis showed that the following packages are required for existing toolchain packages, and should therefore be added to the toolchain:
lua-rpm-macros
lua-srpm-macros
perl-Fedora-VSP
perl-generators
python3-Cython
python3-lxml

###### Change Log  

- Build missing packages in toolchain
- Update toolchain manifests
- Remove unnecessary rebuild of mariner-release, which was built twice
 
#### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline build id: 181034

### CODE REVIEW

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh [hunk 1]
```diff
@@ -419,6 +428,12 @@ chroot_and_install_rpms libxslt
 chroot_and_install_rpms python3
 build_rpm_in_chroot_no_install gtk-doc
 
+# python3-lxml requires python3-Cython and libxslt
+build_rpm_in_chroot_no_install Cython
+chroot_and_install_rpms python3-Cython
+build_rpm_in_chroot_no_install python-lxml
+chroot_and_install_rpms python-lxml
```
Comment (PawelWMS) [inline toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:435]: I think this one is missing in the manifests.

Comment (PawelWMS) [inline toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:435]: I think this one is missing from the manifests.

**Review by PawelWMS — APPROVED** [review]

**Review by dmcilvaney — APPROVED** [review]
Going to do a local build to see if this fixes all the outstanding rebuild issues, but looks good to me here.

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh [hunk 2]
```diff
@@ -264,7 +264,6 @@ chroot_and_install_rpms perl
 build_rpm_in_chroot_no_install flex
 build_rpm_in_chroot_no_install libarchive
 build_rpm_in_chroot_no_install diffutils
-build_rpm_in_chroot_no_install mariner-release
```
Comment (oliviacrain) [inline toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:267]: Good catch on the double-build!

**Review by oliviacrain — APPROVED** [review]

**Review by mfrw — APPROVED** [review]

**Pull request merged**